frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -339,23 +336,40 @@ function ReaderFooter:init()
     if Device:isDesktop() then
         MODE.battery = nil
     end
-    for k, v in pairs(MODE) do
-        mode_tbl[v] = k
-    end
-    for i = 0, #mode_tbl do
-        mode_name = mode_tbl[i]
-        if mode_name then
-            self.mode_index[self.mode_nb] = mode_name
-            self.mode_nb = self.mode_nb + 1
-        end
-    end
+
+    -- self.mode_index will be an array of MODE names, with an additional element
+    -- with key 0 for "off", which feels a bit strange but seems to work...
+    -- (The same is true for self.settings.order which is saved in settings.)
+    self.mode_index = {}
+    self.mode_nb = 0
+
     if self.settings.order then
-        while #self.settings.order < #self.mode_index do
-            self.settings.order[#self.settings.order + 1] = self.mode_index[#self.settings.order + 1]
+        -- Start filling self.mode_index from what's been ordered by the user and saved
+        for i=0, #self.settings.order do
+            local name = self.settings.order[i]
+            -- (if name has been removed from our supported MODEs: ignore it)
+            if MODE[name] then -- this mode still exists
```

Comment:
Is that MODE thing a global?

Reply:
Kind of: a module local variable:
https://github.com/koreader/koreader/blob/03140f505ca475f185bdc197b175ee8a59cf774f/frontend/apps/reader/modules/readerfooter.lua#L24-L42